File: mechanism_of_linux/hello.py
# Pythonプログラムのシステムコール呼び出しの様子を確認するためのサンプル
# $ strace -o hello.py.log python3 ./hello.py
# $ less -N hello.py.log (行数が700行以上になるため、catではなくlessを使用)
print("hello world")

File: mechanism_of_linux/hello.c
/*
 * プロセスが発行するシステムコールを確認するためのサンプル
 * $ strace -o hello.log ./hello
 * $ cat -n hello.log
 */
#include <stdio.h>

int main(void)
{
  puts("hello world");
  return 0;
}
